sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2ScanExec.scala

```diff
@@ -54,27 +53,17 @@ case class DataSourceV2ScanExec(
     Seq(output, source, options).hashCode()
   }
 
-  override def outputPartitioning: physical.Partitioning = readSupport match {
+  override def outputPartitioning: physical.Partitioning = scan match {
```

Comment:
Should `SupportsReportPartitioning` extend `Batch` instead of `Scan`? Then this physical node could just be passed the `Batch` and not the `Scan`, `PartitionReaderFactory`, and partitions.

In fact, I think that this node only requires `output: Seq[AttributeReference], batch: Batch`.

Reply:
If you take my suggesting above to inspect the `Scan` to build the string representation of this node, then I think the arguments should be `scan` and `output`. Then the batch can be fetched here.

For pushedFilters, I think that they should be fetched from the configured scan to build the string representation.